### ISSUE TITLE
Model::create($attributes) doesn't work

### ISSUE THREAD

**Issue by garbin** [darrylkuhn/dialect#5]
$product = Product::create(array(
    'json_data' => array(...)
));

It doesn't work.

**Comment by darrylkuhn** [comment]
Well first thing is that you'd need to set json_data to a string e.g. 

`'json_data' => json_encode(array(...));`

Also check is that you've defined 'json_data' as a json column in the class definition

**Comment by RowdyElectron** [comment]
I'm working on this issue now in L4.2 . Since the jsonColumns declaration can provide a hint as to which columns (thus, model attributes) are json stores, I'm having an easy time visualizing a feasible solution that will take advantage of the Model booting. @darrylkuhn , would you like this to be submitted as a PR? Perhaps you or @garbin would like to test my solution against L5.

**Comment by darrylkuhn** [comment]
@RowdyElectron Thanks for jumping in! What is your approach? If I recall correctly I had to bail on using the boot method as an Eloquent model is technically booted well before data is populated (and therefore all columns are still null upon boot complete).

**Comment by RowdyElectron** [comment]
@darrylkuhn You are totally correct. I wrestled with it quite a bit; trying to cover every edge case. At this moment, there isn't a way I can see to override the default behavior of the models reliably, either through hooking the boot mechanisms or events, that works well with L4 and popular packages. Also, without pre-fetching the schema and inspecting the json/b columns, it is nearly impossible to anticipate a normalized approach to getting/setting json attributes. I can understand why Taylor reduced the Builder to simple 1:1 attribute table operations (aside from creating a common layer for all DBs, but even that doesn't hold true, e.g. SQLLite and Postgres).

In the meantime, I modified the behavior of your lib slightly to make things work for me. My objective was to have an interface to the JSON storage(s) on tables that could be modified with code deployment without requiring a alteration of the db schema, while enforcing good practice. I accomplished this with changing the setJsonAttribute function in the trait:

```
public function setJsonAttribute($attribute, $key, $value)
    {
        /*
         * If column ($attribute) hadn't been defined in the list of attributes,
         * such as when a fresh, non-hydrated, instance of a model is created;
         * inspect the explicitly stated list of json columns
         * */
        /** @noinspection PhpUndefinedFieldInspection */
        if (!isset($this->attributes[$attribute]))
        {
            /*
             * When trying to force a key/value into a column not declared as json,
             * notify the programmer.
             */
            if (false === array_search($attribute, $this->jsonAttributes, true))
            {
                throw new \Exception(
                    'Trying to declare a JSON column exists that hadn\'t' .
                    ' been defined in model\'s list of json columns'
                );
            }

            $this->flagJsonAttribute($key, $attribute);
            $this->{$attribute} = json_encode([$key => $value]);
            return;
        }

        $obj = json_decode($this->{$attribute});
        $obj->$key = $value;
        $this->flagJsonAttribute($key, $attribute);
        $this->{$attribute} = json_encode($obj);
        return;
    }
```

The solution isn't perfect, by any means. This does allow us to use your methodology and let's my guys move forward using json/b goodness in L4. It also has the added benefit of requiring some thought when coding, as I have also mandated the use of model setters/getters, as such:

```
/**
     * Set site name
     *
     * @param string $value
     */
    public function setSiteNameAttribute($value)
    {
        $this->setJsonAttribute('details', 'site_name', $value);
        return;
    }
```

Perhaps this methodology can be expanded and wrapped into your implement. I am curious to know if I have violated any ideas you had had when making this trait by overriding the behavior in setJsonAttribute() .

**Comment by darrylkuhn** [comment]
Hey @RowdyElectron - sorry took a hiatus from this package for a while and now coming back to it. From what I can see you did two things... 
1. Verify that the json column exists in the db schema
2. Added a setter to make it easier to set attributes when the attribute does not yet exist

Do I have it right?
With regard to item 1 - Yes I think that's a good idea and I'll get it into the package...
With regard to item 2 - I've actually added a new "hinting" feature that I think would solve your issue

Let me know what you think
